apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -1312,16 +1787,31 @@ func (s *cloudSuite) TestModifyCloudUpdateAccess(c *gc.C) {
 		},
 	})
 	c.Assert(err, jc.ErrorIsNil)
-	s.backend.CheckCallNames(c, "Cloud", "ControllerTag", "CreateCloudAccess", "GetCloudAccess", "UpdateCloudAccess")
-	s.backend.CheckCall(c, 2, "CreateCloudAccess", "fluffy", names.NewUserTag("fred"), permission.AdminAccess)
-	s.backend.CheckCall(c, 4, "UpdateCloudAccess", "fluffy", names.NewUserTag("fred"), permission.AdminAccess)
 	c.Assert(results.Results, gc.DeepEquals, []params.ErrorResult{
 		{},
 	})
 }
 
 func (s *cloudSuite) TestModifyCloudAlreadyHasAccess(c *gc.C) {
-	s.backend.cloudAccess = permission.AdminAccess
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	cloud := jujucloud.Cloud{
+		Name:      "fluffy",
+		Type:      "fluffy",
+		AuthTypes: []jujucloud.AuthType{jujucloud.EmptyAuthType, jujucloud.UserPassAuthType},
+		Regions:   []jujucloud.Region{{Name: "nether", Endpoint: "endpoint"}},
+	}
+
+	fredTag := names.NewUserTag("fred")
+
+	backend := s.backend.EXPECT()
+	backend.Cloud("fluffy").Return(cloud, nil)
+	backend.CreateCloudAccess("fluffy", fredTag,
+		permission.AdminAccess).Return(errors.New("user already has \"admin\" access or greater")).MaxTimes(1)
+	backend.GetCloudAccess("fluffy", fredTag).Return(permission.AdminAccess,
+		nil).MaxTimes(1)
```

Comment:
Same again here

Reply:
Same case here without the `EXPECT()'s` the test fails. 
```
FAIL: cloud_test.go:1795: cloudSuite.TestModifyCloudAlreadyHasAccess

cloud_test.go:1815:
    results, err := s.api.ModifyCloudAccess(params.ModifyCloudAccessRequest{
        Changes: []params.ModifyCloudAccess{
            {
                Action:   params.GrantCloudAccess,
                CloudTag: names.NewCloudTag("fluffy").String(),
                UserTag:  names.NewUserTag("fred").String(),
                Access:   "admin",
            },
        },
    })
/home/capricorn/code/go/pkg/mod/github.com/golang/mock@v1.6.0/gomock/controller.go:183:
    h.t.Fatalf(format, args...)
... Error: Unexpected call to *mocks.MockBackend.CreateCloudAccess([fluffy user-fred admin]) at /home/capricorn/code/go/src/juju/apiserver/facades/client/cloud/cloud.go:1283 because: there are no expected calls of the method "CreateCloudAccess" for that receiver
```

Reply:
See above also

Reply:
See above. Refactored for the same reasons.